internal/services/applicationinsights/application_insights_resource.go

```diff
@@ -244,7 +235,7 @@ func resourceApplicationInsightsCreateUpdate(d *pluginsdk.ResourceData, meta int
 		Tags:       tags.Expand(d.Get("tags").(map[string]interface{})),
 	}
 
-	_, err := client.ComponentsCreateOrUpdate(ctx, id, insightProperties)
+	_, err = client.ComponentsCreateOrUpdate(ctx, id, insightProperties)
 	if err != nil {
```

Comment:
We can squash this into the condition

```suggestion
	if _, err := client.ComponentsCreateOrUpdate(ctx, id, insightProperties); err != nil {
```